src/streamlink/plugins/n13tv.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import re
+
+from streamlink.compat import urlparse
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import validate
+from streamlink.stream import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+class N13TV(Plugin):
+    url_re = re.compile(r"https?://(?:www\.)?13tv\.co\.il/(live|.*?/)")
+    api_url = "https://13tv-api.oplayer.io/api/getlink/"
+    api_vod_path = "getVideoByFileName"
+    main_js_url_re = re.compile(r'type="text/javascript" src="(.*?main\..+\.js)"')
+    user_id_re = re.compile(r'"data-ccid":"(.*?)"')
+    video_name_re = re.compile(r'"videoRef":"(.*?)"')
+
+    live_schema = validate.Schema([{
+        'Link': validate.url()
+    }])
+
+    vod_schema = validate.Schema([{
+        'ShowTitle': validate.text,
+        'ProtocolType': validate.text,
+        'ServerAddress': validate.text,
+        'MediaRoot': validate.text,
+        'MediaFile': validate.text,
+        'Bitrates': validate.text,
+        'StreamingType': validate.text,
+        'Token': validate.text
+    }])
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return cls.url_re.match(url) is not None
+
+    def _get_streams(self):
+        m = self.url_re.match(self.url)
+        base_path = m and m.group(1)
+        log.debug("Base path = {0}", base_path)
+
+        res = self.session.http.get(self.url)
+
+        if base_path != "live":
+            m = self.video_name_re.search(res.text)
+            video_name = m and m.group(1)
+            log.debug("Video name = {0}", video_name)
```

Comment:
I put it there because the initial ```res = self.session.http.get(self.url)``` contains the info I want for ```video_name```, as well as for ```main_js_path```.  The idea was that if I get it there, I won't have to call ```res = self.session.http.get(self.url)``` again or save the content of that call to use later.  I'm not sure if that makes sense to you.  It did to me.  If you have a suggestion that makes more sense, I'm happy to implement it.  Thanks.

Reply:
No, it's fine. I was just asking because the `base_path` value is compared down below too and `video_name` wasn't used prior to that. If you feel that the HTTP response optimization is necessary, then keep it, not a big deal.

Reply:
It's a minimal optimisation to be fair.  It was just the case that I noticed the way I was doing it before meant I was getting the same page twice, which seemed somewhat unnecessary.